bindings/python/src/boost_python.hpp

```diff
@@ -5,6 +5,12 @@
 #ifndef BOOST_PYTHON_HPP
 #define BOOST_PYTHON_HPP
 
+#ifdef _MSC_VER
+// to work around a mysterious msvc bug (see snprinf defs below) that results in
+// C2039: 'snprintf' is not a member of 'std'
+#include <cstdio>
```

Comment:
it's possible this file isn't included anywhere before ``snprintf()`` is used, but on clang and gcc maybe it got pulled in as a side effect. it should probably be included unconditionally